axelor-base/src/main/java/com/axelor/apps/base/service/pdf/PdfServiceImpl.java

```diff
@@ -0,0 +1,186 @@
+package com.axelor.apps.base.service.pdf;
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.base.db.repo.TraceBackRepository;
+import com.axelor.apps.base.exceptions.BaseExceptionMessage;
+import com.axelor.i18n.I18n;
+import com.axelor.meta.MetaFiles;
+import com.axelor.meta.db.MetaFile;
+import com.axelor.meta.db.repo.MetaFileRepository;
+import com.google.inject.persist.Transactional;
+import com.itextpdf.io.image.ImageData;
+import com.itextpdf.io.image.ImageDataFactory;
+import com.itextpdf.kernel.geom.Rectangle;
+import com.itextpdf.kernel.pdf.PdfDocument;
+import com.itextpdf.kernel.pdf.PdfReader;
+import com.itextpdf.kernel.pdf.PdfWriter;
+import com.itextpdf.kernel.pdf.StampingProperties;
+import com.itextpdf.layout.Document;
+import com.itextpdf.layout.element.Image;
+import com.itextpdf.signatures.BouncyCastleDigest;
+import com.itextpdf.signatures.DigestAlgorithms;
+import com.itextpdf.signatures.IExternalDigest;
+import com.itextpdf.signatures.IExternalSignature;
+import com.itextpdf.signatures.PdfSignatureAppearance;
+import com.itextpdf.signatures.PdfSigner;
+import com.itextpdf.signatures.PrivateKeySignature;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.Security;
+import java.security.cert.Certificate;
+import javax.inject.Inject;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+
+public class PdfServiceImpl implements PdfService {
+
+  protected MetaFiles metaFiles;
+  protected MetaFileRepository metaFileRepository;
+
+  public static final int SIGNATURE_WIDTH = 200;
+  public static final int SIGNATURE_HEIGHT = 100;
+
+  @Inject
+  public PdfServiceImpl(MetaFiles metaFiles, MetaFileRepository metaFileRepository) {
+    this.metaFiles = metaFiles;
+    this.metaFileRepository = metaFileRepository;
+  }
+
+  @Transactional(rollbackOn = {Exception.class})
+  @Override
+  public MetaFile convertImageToPdf(MetaFile metaFile) throws AxelorException {
+    try {
+      if (metaFile == null) {
+        return null;
+      }
+      File tempPdfFile = File.createTempFile(metaFile.getFileName(), ".pdf");
+      FileOutputStream outStream = new FileOutputStream(tempPdfFile);
+      PdfWriter pdfWriter = new PdfWriter(outStream);
+
+      PdfDocument pdfDocument = new PdfDocument(pdfWriter);
```

Comment:
FileOutPutStream must be closed (better to use it inside a try with resource like it's done for document), please check if other created classes must be closed as well